nautobot_netbox_importer/diffsync/netbox.py

```diff
@@ -0,0 +1,491 @@
+        role_wrapper = adapter.wrappers[role_content_type]
+    else:
+        role_wrapper = adapter.configure_model(
+            role_content_type,
+            nautobot_content_type="extras.role",
+            fields={"color": "color", "content_types": "content_types"},
+        )
+
+    def definition(field: SourceField) -> None:
+        field.set_role_importer(role_wrapper)
+
+    return definition
+
+
+def _setup_source() -> SourceAdapter:
+    """Setup NetBox models importers."""
+    adapter = SourceAdapter(name="NetBox")
+    adapter.configure(
+        ignore_fields=(
+            # Following fields fails for dcim.rack, see `./source.py` `_FORCE_FIELDS`
+            "created",
+            "last_updated",
+        ),
+        ignore_content_types={
+            "contenttypes.contenttype": None,  # Nautobot has own content types
+            "sessions.session": None,  # Nautobot has own sessions
+            "dcim.cablepath": None,  # Recreated in Nautobot
+            "admin.logentry": None,  # NetBox 3.0 TBD verify
+            "users.userconfig": None,  # NetBox 3.0 TBD verify
+            "auth.permission": None,  # NetBox 3.0 TBD verify
+        },
+    )
+
+    _setup_base(adapter)
+    _setup_dcim(adapter)
+    _setup_circuits(adapter)
+    _setup_ipam(adapter)
+    _setup_virtualization(adapter)
+
+    return adapter
+
+
+def _setup_base(adapter: SourceAdapter) -> None:
+    adapter.configure_model(
+        "extras.status",
+        identifiers=["name"],
+        default_reference={
+            "name": "Unknown",
+            "content_types": [],
+        },
+    )
+    # TBD: Verify if necessary
+    adapter.configure_model("extras.role")
+    adapter.configure_model(
+        "extras.customfield",
+        fields={
+            "name": "key",
+            "choices": _define_choices,
+        },
+    )
+    adapter.configure_model(
+        "extras.customfieldchoice",
+        fields={
+            "custom_field": "custom_field",
+            "value": "value",
+        },
+    )
+    adapter.configure_model(
+        "extras.taggeditem",
+        fields={
+            "object_id": _define_tagged_object,
+        },
+    )
+    adapter.configure_model(
+        "dcim.locationtype",
+        default_reference={
+            "id": "Unknown",
+            "name": "Unknown",
+            "nestable": True,
+            "content_types": [],
+        },
+    )
+    adapter.configure_model(
+        "dcim.location",
+        fields={
+            "status": "status",
+            "location_type": _define_location_type,
+            "parent": _define_location,
+        },
+    )
+    adapter.configure_model(
+        "auth.user",
+        nautobot_content_type="users.user",
+        identifiers=["username"],
+        fields={
+            "last_login": None,
+            "is_superuser": None,  # NetBox 3.1 TBD verify
+            "password": None,  # NetBox 3.1 TBD verify
+            "user_permissions": None,  # NetBox 3.1 TBD verify
```

Comment:
Permissions are a M2M from ObjectPermission to User not the reverse

Reply:
Created follow-up ticket for this.